plugins/modules/mso_schema_site_anp_epg_useg_attribute.py

```diff
@@ -0,0 +1,264 @@
+    - Use C(true) to set the uSeg Subnet IP address to 0.0.0.0.
+    type: bool
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+notes:
+- Due to restrictions of the MSO REST API concurrent modifications to EPG subnets can be dangerous and corrupt data.
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Add an uSeg attr with attribute_type - ip
+  cisco.mso.mso_schema_template_anp_epg_useg_attribute:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    anp: ANP 1
+    epg: EPG 1
+    site: ansible_test
+    name: useg_attr_ip
+    attribute_type: ip
+    useg_subnet: false
+    value: 10.0.0.0/24
+    state: present
+  delegate_to: localhost
+
+- name: Query a specific EPG uSeg attr with name
+  cisco.mso.mso_schema_template_anp_epg_useg_attribute:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    anp: ANP 1
+    epg: EPG 1
+    name: useg_attr_ip
+    site: ansible_test
+    state: query
+  delegate_to: localhost
+
+- name: Query all EPG uSeg attrs
+  cisco.mso.mso_schema_template_anp_epg_useg_attribute:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    anp: ANP 1
+    epg: EPG 1
+    site: ansible_test
+    state: query
+  delegate_to: localhost
+
+- name: Remove a uSeg attr from an EPG with name
+  cisco.mso.mso_schema_template_anp_epg_useg_attribute:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    anp: ANP 1
+    epg: EPG 1
+    site: ansible_test
+    name: useg_attr_ip
+    state: absent
+  delegate_to: localhost
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.constants import EPG_U_SEG_ATTR_TYPE_MAP
+from ansible_collections.cisco.mso.plugins.module_utils.schema import MSOSchema
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type="str"),
+        site=dict(type="str"),
+        template=dict(type="str"),
+        anp=dict(type="str"),
+        epg=dict(type="str"),
+        name=dict(type="str", aliases=["useg"]),
+        description=dict(type="str", aliases=["descr"]),
+        attribute_type=dict(type="str", aliases=["attr_type"], choices=list(EPG_U_SEG_ATTR_TYPE_MAP.keys())),
+        value=dict(type="str"),
+        operator=dict(type="str", choices=["equals", "contains", "starts_with", "ends_with"]),
```

Comment:
shall we create constant for this list also since in template you also specify it if not mistaken?

Reply:
Sure!